### ISSUE TITLE
EACCES error -13

### ISSUE THREAD

**Issue by umop3plsdn** [rust-lang/rust-analyzer#12911]
Having an issue installing extension I get the following error: 
INFO [7/29/2022, 9:32:25 PM]: Using server binary at /home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu
WARN [7/29/2022, 9:32:25 PM]: /home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu --version: {
  error: [Error: spawnSync /home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu EACCES
  	at Object.spawnSync (node:internal/child_process:1083:20)
  	at spawnSync (node:child_process:821:24)
  	at Yy (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:61:4063)
  	at L0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:3680)
  	at processTicksAndRejections (node:internal/process/task_queues:96:5)
  	at async F0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:1881)
  	at async O0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3977)
  	at async lC (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3808)
  	at async b._activate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
  	at async b._waitForDepsThenActivate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
  	at async b._initialize (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)] {
    errno: -13,
    code: 'EACCES',
    syscall: 'spawnSync /home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu',
    path: '/home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu',
    spawnargs: [ '--version' ]
  },
  status: null,
  signal: null,
  output: null,
  pid: 0,
  stdout: null,
  stderr: null
}
ERROR [7/29/2022, 9:32:25 PM]: Bootstrap error [Error: Failed to execute /home/da5id/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu --version
	at L0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:3922)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async F0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:1881)
	at async O0 (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3977)
	at async lC (/home/da5id/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3808)
	at async b._activate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
	at async b._waitForDepsThenActivate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
	at async b._initialize (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)]

**Comment by umop3plsdn** [comment]
I tried a million other ways of installing to try to get around it including downloading the vsix file.. that failed: 
Installing extensions...
extension/package.json not found inside zip.
Failed Installing Extensions: file:///home/da5id/Downloads/VS-RustAnalyzer.vsix

Then I tried a few other ways to no avail. What ended up working though is git cloning  then running cargo xtask install. HOWEVER, it failed the first time due to no npm being installed so install npm. Anyways, it finally worked.

**Comment by lnicola** [comment]
You should download the VSIX from the GitHub releases page. See also the pinned issue.

**Comment by umop3plsdn** [comment]
ya I had downloaded the vsix from vstudio marketplace guess I should have tried here. Also, this whole time using github and NEVER once did I know the boxes at the top were pinned. I was looking everywhere for a pinned message until I went DUH. They don't really make it that clear honestly. My eyes always skip over those boxes for some reason(NOT ANYMORE). I saw an issue about vscodium and thought it only pertained to that LOL but anways thanks for the reply!
